src/index/import/markdown.ts

```diff
@@ -136,20 +136,30 @@ export function markdownImport(
     ///////////
 
     // All list items in lists. Start with a simple trivial pass.
+    const contentRegex = /^[\t\f\v ]*[\-+\*]\s(\[.\]\s)?/;
```

Comment:
Replacing these with the dataview list matcher which handles various edge cases.